### PR TITLE
add gitlab and other code platform webhook

### PR DESCRIPTION
**Pull request by xiao-jay**
support gitlab，Gitee and codeup platform webhook

### CODE REVIEW

**Comment by rvagg** [comment]
Would you mind doing some formating fixups here to follow the current style? it's a bit hard to follow with so much diff that's just noise. `standard --fix` should help out with that.

I'll have to get Github Actions set up for this so the errors are more obvious because `npm test` should have found this if you ran it locally.

Would you also mind considering how to add tests for these? I know tests are a bit annoying for these kinds of things, but locking in support while the code changes into the future would be helpful so we're not just guessing that this is still working for people using other platforms.

**Comment by xiao-jay** [comment]
ok，@rvagg
thank you for your suggest，i will Resubmit the pr later if i 'm free,and  i will complete formating code and test in  next pr.
In the end，I'm sorry for this sloppy pr.
